svelte-web-frontend/src/lib/ai/motionmetrics/compute-kinematic-motion-descriptors.ts

```diff
@@ -32,28 +37,6 @@ function replaceNaNsWithUndefined(num: number) {
     return isNaN(num) ? undefined : num;
 }
 
-/**
- * Removes duplicate frame times from an array of user poses and corresponding frame times.
- * @param {Pose2DPixelLandmarks[]} userPoses - Array of user poses.
- * @param {number[]} frameTimes - Array of frame times corresponding to the poses.
- * @returns {{ matchingUserPoses: Pose2DPixelLandmarks[], uniqueFrameTimes: number[] }} An object containing adjusted user poses and unique frame times.
- */
-export function removeDuplicateFrameTimes(userPoses: Pose2DPixelLandmarks[], frameTimes: number[]): { matchingUserPoses: Pose2DPixelLandmarks[], uniqueFrameTimes: number[] } {
-    const uniqueFrameTimes = [];
-    const matchingUserPoses = [];
-
-    for (let i = 0; i < frameTimes.length; i++) {
-        const currentTime = frameTimes[i];
-
-        // Check if the current frame time is not the same as the previous one
-        if (i === 0 || currentTime !== frameTimes[i - 1]) {
-            uniqueFrameTimes.push(currentTime);
-            matchingUserPoses.push(userPoses[i]);
-        }
-    }
-
-    return { matchingUserPoses, uniqueFrameTimes };
-}
 
 /**
  * Calculates the MAE and RMSE for velocity, acceleration, and jerk between matchingUserPoses and referencePoses.
```

Comment:
I think you should create a test for the 3d kinematic motion descriptors as well. We can walk through how to do that in our meeting tomorrow.